Model/Payment/FastPayment.php

```diff
@@ -236,15 +236,14 @@ public function capture(InfoInterface $payment, $amount)
      */
     public function refund(InfoInterface $payment, $amount)
     {
-        $transactionId = $this->fastCheckoutHelper->guid() . '-' . Transaction::TYPE_REFUND;
         $parentTransactionId = $payment->getParentTransactionId();
-        $this->fastCheckoutHelper->log('transaction id in refund: ' . $transactionId, Logger::DEBUG);
+        $this->fastCheckoutHelper->log('parent transaction id in refund: ' . $parentTransactionId, Logger::DEBUG);
         $order = $payment->getOrder();
         if ($order->getFastOrderId()) {
             try {
                 $this->fastCheckoutHelper->log("in before Refund", Logger::DEBUG);
                 // send fast the refund request
-                $this->sendRefund($order, $payment, $amount, $transactionId);
+                $transactionId = $this->sendRefund($order, $payment, $amount);
```

Comment:
Should the line 252 be inside this ? Not sure if it has acess to transactionId ?

Reply:
It should be fine where it is. It's still in the same scope even though it's defined inside the `try{}`.